CMakeLists.txt

```diff
@@ -107,6 +108,10 @@ if (WITH_C_API AND WITH_PYTHON)
     "different Python interpreter from compiling.")
 endif()
 
+if (WITH_C_API)
```

Comment:
How about we rename WITH_C_API into PADDLE_BUILD_INFERENCE_LIB ?

Reply:
As WITH_C_API is used for a long time in old PaddlePaddle, users are familiar with it, we may remain it. And for fluid, we will add WITH_FLUID option for both training and inference. (see https://github.com/PaddlePaddle/Paddle/issues/7578#issuecomment-358191146)